charts/consul/test/terraform/eks/main.tf

```diff
@@ -84,4 +87,93 @@ data "aws_eks_cluster" "cluster" {
 data "aws_eks_cluster_auth" "cluster" {
   count = var.cluster_count
   name  = module.eks[count.index].cluster_id
+}
+
+# The following resources are only applied when cluster_count=2 to set up vpc peering and the appropriate routes and
+# security groups so traffic between vpcs is allowed. There is validation to ensure cluster_count can be 1 or 2.
+
+# Each EKS cluster needs to allow ingress traffic from the other VPC.
+resource "aws_security_group_rule" "allowingressfrom1-0" {
+  count             = var.cluster_count > 1 ? 1 : 0
+  type              = "ingress"
+  from_port         = 0
+  to_port           = 65535
+  protocol          = "tcp"
+  cidr_blocks       = [module.vpc[1].vpc_cidr_block]
+  security_group_id = module.eks[0].cluster_primary_security_group_id
+}
+
+resource "aws_security_group_rule" "allowingressfrom0-1" {
+  count             = var.cluster_count > 1 ? 1 : 0
+  type              = "ingress"
+  from_port         = 0
+  to_port           = 65535
+  protocol          = "tcp"
+  cidr_blocks       = [module.vpc[0].vpc_cidr_block]
+  security_group_id = module.eks[1].cluster_primary_security_group_id
+}
+
+# Create a peering connection. This is the requester's side of the connection.
+resource "aws_vpc_peering_connection" "peer" {
+  count         = var.cluster_count > 1 ? 1 : 0
+  vpc_id        = module.vpc[0].vpc_id
+  peer_vpc_id   = module.vpc[1].vpc_id
+  peer_owner_id = data.aws_caller_identity.caller.account_id
+  peer_region   = var.region
+  auto_accept   = false
+
+  tags = {
+    Side = "Requester"
+  }
+}
+
+# Accepter's side of the vpc peering connection.
+resource "aws_vpc_peering_connection_accepter" "peer" {
+  count                     = var.cluster_count > 1 ? 1 : 0
+  vpc_peering_connection_id = aws_vpc_peering_connection.peer[0].id
+  auto_accept               = true
+
+  tags = {
+    Side = "Accepter"
+  }
+}
+
+# Get the route table ids that are associated with eks cluster 0, so we can create routes in those route tables.
+data "aws_route_tables" "rtseks0" {
+  count  = var.cluster_count > 1 ? 1 : 0
+  vpc_id = module.vpc[0].vpc_id
+
+  filter {
+    name   = "tag:Name"
+    values = [format("%s-*", module.eks[0].cluster_id)]
+  }
+}
+
+# Get the route table ids that are associated with eks cluster 1, so we can create routes in those route tables.
+data "aws_route_tables" "rtseks1" {
+  count  = var.cluster_count > 1 ? 1 : 0
+  vpc_id = module.vpc[1].vpc_id
+
+  filter {
+    name   = "tag:Name"
+    values = [format("%s-*", module.eks[1].cluster_id)]
+  }
+}
```

Comment:
will the below work? or will it interact poorly with the `aws_route` in the next step? it could potentially work as rtseks[0]
```suggestion
# Get the route table ids that are associated with eks cluster 0, so we can create routes in those route tables.
data "aws_route_tables" "rtseks" {
  count  = var.cluster_count > 1 ? 2 : 0
  vpc_id = module.vpc[count.index].vpc_id

  filter {
    name   = "tag:Name"
    values = [format("%s-*", module.eks[count.index].cluster_id)]
  }
}
```

Reply:
oh yeah that could possibly work! but I think i'll end up removing these per iryna's suggestion to use the vpc module to grab the route tables instead!

Reply:
sounds good!! 👍